ADR/ADR044-translate-api-response-at-request-time.md

```diff
@@ -0,0 +1,417 @@
+      },
+      "routing_conditions": []
+    }
+  ]
+}
+```
+
+Or with translated fields stored as objects (as proposed in the unimplemented [ADR007: Store form text data as objects](ADR007-store-form-text-data-as-objects.md))
+
+```jsonc
+{
+  "form_id": "7",
+  "name": {
+    "en": "A form for testing file upload",
+    "cy": "Ffurflen ar gyfer profi uwchlwytho ffeiliau"
+  },
+  "submission_email": "david.biddle@digital.cabinet-office.gov.uk",
+  "privacy_policy_url": null,
+  "form_slug": "a-form-for-testing-file-upload",
+  "support_email": null,
+  "support_phone": null,
+  "support_url": null,
+  "support_url_text": null,
+  "declaration_text": null,
+  "question_section_completed": false,
+  "declaration_section_completed": false,
+  "created_at": "2025-03-27T12:05:49.395300Z",
+  "updated_at": "2025-07-29T14:35:44.423406Z",
+  "creator_id": 1,
+  "what_happens_next_markdown": null,
+  "payment_url": null,
+  "submission_type": "email",
+  "share_preview_completed": false,
+  "s3_bucket_name": null,
+  "s3_bucket_aws_account_id": null,
+  "s3_bucket_region": null,
+  "language": "en",
+  "start_page": 33,
+  "steps": [
+    {
+      "id": 33,
+      "position": 1,
+      "next_step_id": null,
+      "type": "question_page",
+      "data": {
+        "question_text": {
+          "en": "Upload your file",
+          "cy": "Llwythwch eich ffeil i fyny"
+        },
+        "hint_text": "",
+        "answer_type": "file",
+        "is_optional": false,
+        "answer_settings": null,
+        "page_heading": {
+          "en": "Annual report",
+          "cy": "Adroddiad blynyddol"
+        },
+        "guidance_markdown": {
+          "en": "Some guidance",
+          "cy": "Peth arweiniad"
+        },
+        "is_repeatable": false
+      },
+      "routing_conditions": []
+    }
+  ]
+}
+```
+
+These cases are essentially the same in how forms-runner deals with them.
+Benefits:
+
+- Few admin/api changes required
+  Disadvantages:
+- More runner logic, since we'd have to manage choosing the right translation and fallback behaviour in runner.
+- We'd get less benefit from the Mobility gem - we'd be using it for managing the columns and dealing with translations in admin, but we wouldn't be using its querying or fallback features in forms-runner.
+- For forms with lots of content (e.g. multiple pages with a large amount of detailed guidance or what happens next text), this would increase the size of the request quite a lot since this information would be present in two different languages. Much of that content wouldn't be used for a given request, since the runner only shows one language at a time.
+- Additional languages would make it even more verbose.
+
+### 2. Selecting languages at request time
+
+In this approach, we would allow the runner to request a specific language when requesting the form from the API, and the API would only supply translations for the requested language.
+
+So `/forms/7/live?locale=en` would return:
```

Comment:
It's a little confusing that the URL parameter here is called `locale` while the attribute in the form document is called `language`; can we make the two consistent?

Reply:
Hmm... I guess that makes sense for what's stored in the database, but for a form document response that has one translation in it I think it would be helpful for it to be tagged with the translation language; if only for debugging purposes!